stix2/environment.py

```diff
@@ -66,3 +73,84 @@ def create(self, cls, **kwargs):
             properties.update(**kwargs)
 
         return cls(**properties)
+
+
+class Environment(object):
+    """
+
+    Args:
+        factory (ObjectFactory, optional): Factory for creating objects with common
+            defaults for certain properties.
+        store (DataStore, optional): Data store providing the source and sink for the
+            environment.
+        source (DataSource, optional): Source for retrieving STIX objects.
+        sink (DataSink, optional): Destination for saving STIX objects.
+            Invalid if `store` is also provided.
+    """
+
+    def __init__(self, factory=ObjectFactory(), store=None, source=None, sink=None):
+        self.factory = factory
+        self.source = CompositeDataSource()
+        if store:
+            self.source.add_data_source(store.source)
+            self.sink = store.sink
+        if source:
+            self.source.add_data_source(source)
+        if sink:
+            if store:
+                raise ValueError("Data store already provided! Environment may only have one data sink.")
+            self.sink = sink
+
+    def create(self, *args, **kwargs):
+        return self.factory.create(*args, **kwargs)
+    create.__doc__ = ObjectFactory.create.__doc__
+
+    def get(self, *args, **kwargs):
+        try:
+            return self.source.get(*args, **kwargs)
+        except AttributeError:
+            raise AttributeError('Environment has no data source to query')
+    get.__doc__ = DataStore.get.__doc__
```

Comment:
I feel like these should use the `DataSource` docstrings, but after taking a look at those, they could probably be cleaned up and reorganized between `DataStore` and `DataSource`. It's fine for now. I created #53 to address that.